src/main/kotlin/net/atos/zac/admin/ZaakafhandelParameterBeheerService.kt

```diff
@@ -136,36 +138,38 @@ class ZaakafhandelParameterBeheerService @Inject constructor(
         val zaaktype = ztcClientService.readZaaktype(zaaktypeUri)
         if (!zaaktype.concept) {
             val omschrijving = zaaktype.omschrijving
-            val vorigeZaakafhandelparameters = readRecentsteZaakafhandelParameters(omschrijving)
-            val nieuweZaakafhandelParameters = ZaakafhandelParameters().apply {
+            val previousZaakafhandelparameters = readRecentsteZaakafhandelParameters(omschrijving)
+            val newZaakafhandelParameters = ZaakafhandelParameters().apply {
```

Comment:
Just playing around, I think this might work (with some more of my comments):
```kotlin
            readRecentsteZaakafhandelParameters(zaaktype.omschrijving)
                .let {
                    it.id = null // detach the entity

                    // set the new case type
                    val previousZaakTypeUUID = it.zaakTypeUUID
                    with(zaaktype) {
                        it.zaakTypeUUID = url.extractUuid()
                        it.zaaktypeOmschrijving = omschrijving
                        it.einddatumGeplandWaarschuwing = servicenorm?.run {
                            it.einddatumGeplandWaarschuwing
                        }
                    }
                    // clone collections
                    it.setHumanTaskParametersCollection(it.humanTaskParametersCollection.clone())
                    it.setUserEventListenerParametersCollection(it.userEventListenerParametersCollection.clone())
                    it.setMailtemplateKoppelingen(it.mailtemplateKoppelingen.clone())
                    // remap zaakbeeindiggegevens
                    remapZaakbeeindigGegevens(it, zaaktype)
                    // copy smart documents data
                    smartDocumentsTemplatesService.storeTemplatesMapping(
                        smartDocumentsTemplatesService.getTemplatesMapping(previousZaakTypeUUID), 
                        it.zaakTypeUUID
                    )
                    // keep the other values
                    it
                }
                .run(::createZaakafhandelParameters)
```

Reply:
Maybe extract the code in the let into a new extension method:
```kotlin
    private fun ZaakafhandelParameters.cloneWith(zaaktype: ZaakType) =
        this.let {
            it.id = null // detach the entity

            // set the new case type
            val previousZaakTypeUUID = it.zaakTypeUUID
            with(zaaktype) {
                it.zaakTypeUUID = url.extractUuid()
                it.zaaktypeOmschrijving = omschrijving
                it.einddatumGeplandWaarschuwing = servicenorm?.run {
                    it.einddatumGeplandWaarschuwing
                }
            }
            // clone collections
            it.setHumanTaskParametersCollection(it.humanTaskParametersCollection.clone())
            it.setUserEventListenerParametersCollection(it.userEventListenerParametersCollection.clone())
            it.setMailtemplateKoppelingen(it.mailtemplateKoppelingen.clone())
            // remap zaakbeeindiggegevens
            remapZaakbeeindigGegevens(it, zaaktype)
            // copy smart documents data
            smartDocumentsTemplatesService.storeTemplatesMapping(
                smartDocumentsTemplatesService.getTemplatesMapping(previousZaakTypeUUID),
                it.zaakTypeUUID
            )
            // keep the other values
            it
        }
```